app/services/category_hashtag_data_source.rb

```diff
@@ -52,11 +52,43 @@ def self.search(guardian, term, limit)
   end
 
   def self.search_sort(search_results, term)
-    search_results
-      .select { |item| item.slug == term }
-      .sort_by { |item| item.text.downcase }
-      .concat(
-        search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
-      )
+    if term.present?
+      search_results
+        .select { |item| item.slug == term }
+        .sort_by { |item| item.text.downcase }
+        .concat(
+          search_results.select { |item| item.slug != term }.sort_by { |item| item.text.downcase },
+        )
+    else
+      search_results.sort_by { |item| item.text.downcase }
+    end
+  end
+
+  def self.search_without_term(guardian, limit)
+    guardian_categories = Site.new(guardian).categories
+
+    category_id_sql = <<~SQL
+      SELECT category_id, MAX(posts.created_at)
+      FROM topics
+      INNER JOIN posts ON posts.topic_id = topics.id
+      WHERE topics.deleted_at IS NULL
+        AND posts.deleted_at IS NULL
+        AND posts.created_at > (NOW() - INTERVAL '2 WEEKS')
+        AND topics.category_id IN (:category_ids)
+      GROUP BY category_id
+      ORDER BY MAX(posts.created_at) DESC
+      LIMIT :limit
+    SQL
+    category_ids =
+      DB.query(
+        category_id_sql,
+        category_ids: guardian_categories.map { |cat| cat[:id] },
+        limit: limit,
+      ).map(&:category_id)
+
+    guardian_categories
+      .select { |category| category_ids.include?(category[:id]) }
+      .take(limit)
+      .map { |category| category_to_hashtag_item(guardian_categories, category) }
```

Comment:
Based on the implementation of the method, I'm a little confused about why we need to pass in all the `guardian_categories` instead of just passing the parent category, which we can preload, to the method.

Reply:
Ah yeah you are right, will just change it to pass the parent category instead.